core/src/main/java/org/apache/accumulo/core/file/blockfile/impl/CachableBlockFile.java

```diff
@@ -83,8 +84,24 @@ public CachableBuilder conf(Configuration hadoopConf) {
     }
 
     public CachableBuilder fsPath(FileSystem fs, Path dataFile) {
+      return fsPath(fs, dataFile, false);
+    }
+
+    public CachableBuilder fsPath(FileSystem fs, Path dataFile, boolean dropCacheBehind) {
       this.cacheId = pathToCacheId(dataFile);
-      this.inputSupplier = () -> fs.open(dataFile);
+      this.inputSupplier = () -> {
+        FSDataInputStream is = fs.open(dataFile);
+        if (dropCacheBehind) {
+          // Tell the DataNode that the write ahead log does not need to be cached in the OS page
+          // cache
+          try {
+            is.setDropBehind(Boolean.TRUE);
+          } catch (IOException | UnsupportedOperationException e) {
+            log.debug("setDropBehind not enabled for wal file: {}", dataFile);
```

Comment:
I'm wondering if it's useful to show the message from the exception in this debug message, at least in the case of IOException.

Reply:
I can do that, do you think in the case of IOException, that it should be higher than debug? warn maybe?

Reply:
I'm not really sure what circumstances could trigger this. It's probably not worth logging higher than debug, since this is really an internal performance optimization, and not a scenario the user really needs to concern themselves with unless they're trying to eek out every nanosecond of performance, in which case they can just look at the debug logs.

Reply:
> I can do that, do you think in the case of IOException, that it should be higher than debug? warn maybe?

Hopefully if the stream is in a bad state and it throws an IOE here then it will throw an IOE later when trying to use it.  Those later throws should log something, so maybe the debug here is ok?  Would be nice to understand when it throws an IOE.